modules/siddhi-launcher/src/main/java/org/wso2/siddhi/sdk/launcher/LauncherUtils.java

```diff
@@ -0,0 +1,149 @@
+ */
+public class LauncherUtils {
+
+    private static final Logger log = Logger.getLogger(Main.class);
+
+    public static void runProgram(boolean isDebugEnabled, String[] args) {
+
+        if (args.length == 0 || args[0] == null) {
+            throw new FileReadException("No Siddhi App file Path given");
+        }
+        String siddhiAppPath = args[0];
+        // Validate siddhiApp
+        String siddhiApp = validateAndGetSiddhdiApp(siddhiAppPath);
+        String inputFilePath;
+        String inputFile = "";
+        if (!(args.length == 1 || args[1] == null || args[1].equalsIgnoreCase(""))) {
+            inputFilePath = args[1];
+            try {
+                inputFile = readText(inputFilePath);
+            } catch (IOException e) {
+                throw new FileReadException("Failed to read event input file:" + inputFilePath + ":" + e);
+            }
+        } else {
+            log.info("Event Input file is not provided or file is empty");
+        }
+
+        if (!siddhiApp.equals("")) {
+            if (!isDebugEnabled) {
+                try {
+                    SiddhiRun siddhiRun = new SiddhiRun();
+                    siddhiRun.runSiddhi(siddhiApp, inputFile);
+                } catch (InterruptedException e) {
+                    throw new FileReadException("Siddhi App execution error: " + e);
+                }
+            } else {
+                VMDebugManager vmDebugManager = VMDebugManager.getInstance();
+                vmDebugManager.mainInit(siddhiAppPath, siddhiApp, inputFile);
+            }
+        } else {
+            throw new FileReadException("No valid SiddhiApp found in the file");
+        }
+    }
+
+    /**
+     * Validates the Siddhi App path.
+     *
+     * @param siddhiAppPath path to the siddhiApp file
+     */
+    private static boolean validateSiddhiAppPath(String siddhiAppPath) {
+        File siddhiAppFile = new File(siddhiAppPath);
+        String fileName = siddhiAppFile.getName();
+        String extension = fileName.substring(fileName.lastIndexOf(".") + 1);
+        if (!siddhiAppFile.exists() || !siddhiAppFile.isFile()) {
+            throw new FileReadException("Invalid siddhi app file path:" + siddhiAppPath);
+        }
+        if (!extension.equalsIgnoreCase("siddhi")) {
+            throw new FileReadException("Invalid siddhi file extension detected in the file:" + extension);
+        }
+        return true;
+    }
+
+    /**
+     * Validates the Siddhi App.
+     *
+     * @param siddhiAppPath path to the siddhiApp file
+     */
+    private static String validateAndGetSiddhdiApp(String siddhiAppPath) {
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String siddhiApp = "";
+        boolean isValidSiddhiAppPath = validateSiddhiAppPath(siddhiAppPath);
+        if (isValidSiddhiAppPath) {
+            try {
+                siddhiApp = readText(siddhiAppPath);
+                SiddhiAppRuntime siddhiAppRuntime = null;
```

Comment:
You do not need this null assignment

Reply:
Otherwise in finally block I cannot shutdown the siddhiAppRuntime. It will show an error saying "siddhiAppRuntime" might not have been initialized.